uaclient/entitlements/livepatch.py

```diff
@@ -135,50 +232,51 @@ def _perform_enable(self) -> bool:
 
         @return: True on success, False otherwise.
         """
-        if not util.which("/snap/bin/canonical-livepatch"):
-            if not util.which(SNAP_CMD):
-                print("Installing snapd")
-                print(status.MESSAGE_APT_UPDATING_LISTS)
-                try:
-                    apt.run_apt_command(
-                        ["apt-get", "update"], status.MESSAGE_APT_UPDATE_FAILED
-                    )
-                except exceptions.UserFacingError as e:
-                    logging.debug(
-                        "Trying to install snapd."
-                        " Ignoring apt-get update failure: %s",
-                        str(e),
-                    )
-                util.subp(
-                    ["apt-get", "install", "--assume-yes", "snapd"],
-                    capture=True,
-                    retry_sleeps=apt.APT_RETRIES,
-                )
-            elif "snapd" not in apt.get_installed_packages():
-                raise exceptions.UserFacingError(
-                    "/usr/bin/snap is present but snapd is not installed;"
-                    " cannot enable {}".format(self.title)
-                )
-
+        if not util.which(SNAP_CMD):
```

Comment:
Notice that I did refactor this section a little bit to ensure that we always try to configure the snap proxy if necessary (before it would only install the snap proxy if canonical-livepatch was not installed).

I basically just moved the `if not util.which(SNAP_CMD)` out from underneath the `if not util.which("/snap/bin/canonical-livepatch")`

Reply:
Makes sense to me because we are trying to enable livepatch at this point anyway +1 not nesting that logic unnecessarily. As you mentioned, if livepatch were already installed, accidentally **not** setting up the snap proxy settings is a gap that needed addressing.